Avoid/deprecate usages of RecipeChoice#getItemStack()

qixils opened the pull request:
Replaces some internal usages of this method with the more proper `RecipeChoice#test(ItemStack)`. Every other method that utilizes `RecipeChoice#getItemStack()` has been deprecated with a message suggesting the modern equivalent that should be used instead.

kennytv commented on patches/api/0175-Fix-Spigot-annotation-mistakes.patch:
```diff
@@ -524,6 +524,20 @@ index 6cb2bf60fb970b646451c02faf71a14a80e6a56b..7cf8ffb5a29d1cf382064b3aca0353a5
+      * Get the input material.
+      *
+      * @return The input material.
++     * @deprecated Use {@link #getInputChoice()} instead for more complete data. <!-- Paper -->
```
No need for `Paper` comments in docs

qixils replied:
👍🏻

qixils committed:
Avoid usages of RecipeChoice#getItemStack()

Replaces some internal usages of this method with RecipeChoice#test(ItemStack) and deprecates every other method still utilizing this legacy method.
